atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/cache/CacheStoreImplTest.java

```diff
@@ -66,6 +70,10 @@ public void multipleCallsToGetReturnsTheSameCache() {
         TransactionScopedCache cache2 = cacheStore.getCache(TIMESTAMP_2);
 
         assertThat(cacheStore.getCache(TIMESTAMP_1)).isEqualTo(cache1).isNotEqualTo(cache2);
+
+        ArgumentCaptor<Gauge<Integer>> cacheInstanceCount = ArgumentCaptor.forClass(Gauge.class);
+        verify(metrics).setCacheInstanceCountGauge(cacheInstanceCount.capture());
+        assertThat(cacheInstanceCount.getValue()).extracting(Gauge::getValue).isEqualTo(2);
```

Comment:
Can you extract out a method to do this, then add this check throughout all of these tests? It seems like a nice idea, to be able to confirm how many caches we have stored in memory.